From df0831ce57dc28b8cee9133378fb0f23977fc44c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Tue, 16 Jan 2018 11:25:29 +0000
Subject: meson: fix check whether both gles2 and opengl headers can be
 included

cc.compiles() doesn't support the prefix: kwarg currently, so it
never had any effect.

https://github.com/mesonbuild/meson/issues/2364
https://bugzilla.gnome.org/show_bug.cgi?id=787964
---
 gst-libs/gst/gl/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/gl/meson.build b/gst-libs/gst/gl/meson.build
index 9762a09..68af27f 100644
--- a/gst-libs/gst/gl/meson.build
+++ b/gst-libs/gst/gl/meson.build
@@ -366,7 +366,9 @@ endif
 # can we include both gles2 and opengl headers?
 if gles2_dep.found() and gl_dep.found()
   gl_include_block = gl_include_header + gles_includes + opengl_includes
-  if not cc.compiles('void f (void) {}', prefix : gl_include_block, dependencies : [gles2_dep, gl_dep] )
+  # TODO: Revert to passing gl_include_block via prefix: once
+  # https://github.com/mesonbuild/meson/issues/2364 is fixed
+  if not cc.compiles(gl_include_block + '\n' + 'void f (void) {}',dependencies : [gles2_dep, gl_dep])
     message ('Cannot include both OpenGL and OpenGL ES headers')
     if need_gles2 != 'yes'
       gles2_dep = unneeded_dep
-- 
cgit v1.1

